src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Bindings/StorageBlobContainerExtensions.cs

```diff
@@ -48,7 +49,8 @@ internal static class StorageBlobContainerExtensions
 
                 if (result == null || result.HttpStatusCode != 404)
                 {
-                    throw;
+                    ExceptionDispatchInfo.Capture(exception).Throw();
```

Comment:
Here and in all the other instances below you're doing an explicit recapture/throw for exceptions that we're already using the "throw" keyword for. My understanding is that as long as we're not doing a rethrow of the form "throw ex", the stack trace is maintained already without a recapture.

It seems we're already doing this correctly in most places. E.g. in places in our execution pipeline where we catch an invocation exception and don't throw immediately, we're capturing and throwing the ExceptionDispatchInfo and using it to rethrow later, e.g. [here](https://github.com/Azure/azure-webjobs-sdk/blob/f7da27d133b9ce0aadf9918770f6ed45748f3c8a/src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs#L134).

Reply:
Thanks for your response, @mathewc .  This statement is not quite correct: 

> as long as we're not doing a rethrow of the form "throw ex", the stack trace is maintained already without a recapture.

Please see my example in the issue https://github.com/Azure/azure-webjobs-sdk/issues/2445 .
The stack you get with `throw` shows the line where the exception came up in the `try`-block, bit not the line where it was re-thrown in the `catch`-block. The stack you get with `ExceptionDispatchInfo` shows both. Please see the above-linked post for an example.